### PR TITLE
Added audiobook support

### PR DESCRIPTION
**Pull request by modembug**
Adds support for audiobooks. Includes functionality to sync progress of built in player to bookmark entry. This allows users to listen on multiple platforms and maintain position.

image

image

### CODE REVIEW

File: cps/web.py [hunk 1]
```diff
@@ -990,7 +991,7 @@ def get_update_status():
     status = {}
     if request.method == "GET":
         # should be automatically replaced by git with current commit hash
-        commit_id = '$Format:%H$'
+        commit_id = '66c1966b44454828b0dd431c2d07c5b71c064598'
```
Comment (davydov-vyacheslav) [inline cps/web.py]: I'm new person in this person and mostly as end-user (sorry, I don't know python). Why this changed? (and line 2472 as well)

File: cps/templates/listenmp3.html [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  <link rel="apple-touch-icon" sizes="140x140" href="{{ url_for('static', filename='favicon.ico') }}">
+  <link rel="shortcut icon" href="{{ url_for('static', filename='favicon.ico') }}">
+  <link href="{{ url_for('static', filename='css/libs/bootstrap.min.css') }}" rel="stylesheet" media="screen">
+  <link rel="stylesheet" href="{{ url_for('static', filename='css/libs/bar-ui.css') }}" />
+
+  <!-- <link rel="stylesheet" href="{{ url_for('static', filename='css/libs/normalize.css') }}">
+  <link rel="stylesheet" href="{{ url_for('static', filename='css/main.css') }}">
+  <link rel="stylesheet" href="{{ url_for('static', filename='css/popup.css') }}"> -->
+
+  <style type="text/css">
```
Comment (davydov-vyacheslav) [inline cps/templates/listenmp3.html]: could you extract styles to separate file?

**Comment by davydov-vyacheslav** [comment]
Just wondering, is your implementation stream mp3 or loads the file fully and then allows to play it?

File: cps/templates/detail.html [hunk 3]
```diff
@@ -46,12 +46,18 @@
               {% if entry.data|length  %}
               <div class="btn-group" role="group">
                 <button id="read-in-browser" type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                  <span class="glyphicon glyphicon-eye-open"></span> {{_('Read in browser')}}
+                  {% for format in entry.data %}
+                    {% if format.format|lower == 'mp3' %}
```
Comment (davydov-vyacheslav) [inline cps/templates/detail.html]: what about aax, m4a ?  (in /templates/index.html as well)

File: cps/templates/listenmp3.html (hunk 2 above)
Comment (modembug) [reply cps/templates/listenmp3.html]: Yeah, or you can if you want :)

**Comment by modembug** [comment]
This implementation streams the content.

File: cps/web.py (hunk 1 above)
Comment (OzzieIsaacs) [reply cps/web.py]: Both lines are used by git during commit/push and replaced automatically with the current commit stamp. To make the auto update working, it should be the original text. But I can change this on merge.

File: cps/templates/listenmp3.html (hunk 2 above)
Comment (OzzieIsaacs) [reply cps/templates/listenmp3.html]: I agree with @davydov-vyacheslav. Please move the styles to a separate file

File: cps/templates/listenmp3.html [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        filePath: "{{ url_for('static', filename='js/libs/') }}",
+        cssPath: "{{ url_for('static', filename='css/') }}",
+        bookUrl: "{{ url_for('static', filename=mp3file) }}/",
+        bookmarkUrl: "{{ url_for('bookmark', book_id=mp3file, book_format='MP3') }}",
+        bookmark: "{{ bookmark.bookmark_key if bookmark != None }}",
+        useBookmarks: {{ g.user.is_authenticated | tojson }}
+            };
+    </script>
+
+</body>
```
Comment (OzzieIsaacs) [inline cps/templates/listenmp3.html]: My firefox is complaining about thte closing body at this line, there might be a div missing or to much, could you please check this.

Comment (OzzieIsaacs) [reply cps/templates/listenmp3.html]: I'm having a problem with the size of the graphic, on my computer it looks like this:
bildschirmfoto_2018-06-02_18-04-30
Took me some time to figure out that it is working, and the controller is in the hidden area (where you can't scroll to)

**Comment by Mohammed90** [comment]
Is this PR missing anything for the functionality beyond the comments mentioned and conflict reslution?
- Move CSS to corresponding file
- Missing closing div tag
- Correcting image path (to cps/static/css/images)

**Comment by modembug** [comment]
Works fine for me still.
________________________________

**Comment by Mohammed90** [comment]
Great, then it's just the few requested cleanups. Could you do them, or should I take it up? I don't have permissions to your own PR/fork, so it'll have to be another PR.

**Comment by bjhiggins** [comment]
@modembug - could you please grant @Mohammed90 some sort of access to the calibre-web Pull Request so that he may finish it up so that @janeczku will merge it into the master code branch?

Thank you - I've been patiently waiting for this feature in calibre-web!

**Comment by bodybybuddha** [comment]
@modembug - You may want to close out this PR.  Thanks in advance!

**Comment by OzzieIsaacs** [comment]
I merged the  bodybybuddha PR into the development branch. I'm working on refactoring the code and will make it then available to the public. 
Thanks a lot for the patience and the work

**Pull request closed without merging**